packages/shorebird_cli/test/src/commands/collaborators/delete_collaborators_command_test.dart

```diff
@@ -0,0 +1,176 @@
+import 'package:args/args.dart';
+import 'package:http/http.dart' as http;
+import 'package:mason_logger/mason_logger.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:shorebird_cli/src/auth/auth.dart';
+import 'package:shorebird_cli/src/commands/commands.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+import 'package:test/test.dart';
+
+class _MockArgResults extends Mock implements ArgResults {}
+
+class _MockHttpClient extends Mock implements http.Client {}
+
+class _MockAuth extends Mock implements Auth {}
+
+class _MockCodePushClient extends Mock implements CodePushClient {}
+
+class _MockLogger extends Mock implements Logger {}
+
+class _MockProgress extends Mock implements Progress {}
+
+void main() {
+  group('delete', () {
+    const appId = 'test-app-id';
+    const email = 'jane.doe@shorebird.dev';
+    const collaborator = Collaborator(userId: 0, email: email);
+
+    late ArgResults argResults;
+    late http.Client httpClient;
+    late Auth auth;
+    late CodePushClient codePushClient;
+    late Logger logger;
+    late Progress progress;
+    late DeleteCollaboratorsCommand command;
+
+    setUp(() {
+      argResults = _MockArgResults();
+      httpClient = _MockHttpClient();
+      auth = _MockAuth();
+      codePushClient = _MockCodePushClient();
+      logger = _MockLogger();
+      progress = _MockProgress();
+      command = DeleteCollaboratorsCommand(
+        auth: auth,
+        buildCodePushClient: ({
+          required http.Client httpClient,
+          Uri? hostedUri,
+        }) {
+          return codePushClient;
+        },
+        logger: logger,
+      )..testArgResults = argResults;
+
+      when(() => argResults['app-id']).thenReturn(appId);
+      when(() => argResults['email']).thenReturn(email);
+      when(() => auth.isAuthenticated).thenReturn(true);
+      when(() => auth.client).thenReturn(httpClient);
+      when(() => logger.confirm(any())).thenReturn(true);
+      when(() => logger.progress(any())).thenReturn(progress);
+      when(
+        () => codePushClient.getCollaborators(appId: any(named: 'appId')),
+      ).thenAnswer((_) async => [collaborator]);
+      when(
+        () => codePushClient.deleteCollaborator(
+          appId: any(named: 'appId'),
+          userId: any(named: 'userId'),
+        ),
+      ).thenAnswer((_) async {});
+    });
+
+    test('description is correct', () {
+      expect(
+        command.description,
+        equals('Delete an existing collaborator from a Shorebird app.'),
+      );
+    });
+
+    test('returns ExitCode.noUser when not logged in', () async {
+      when(() => auth.isAuthenticated).thenReturn(false);
+      expect(await command.run(), ExitCode.noUser.code);
+    });
+
+    test('returns ExitCode.usage when app id is missing.', () async {
+      when(() => argResults['app-id']).thenReturn(null);
```

Comment:
Are we testing fetching the appId from `shorebird.yaml`?

Reply:
good catch fixed in #511